From 7130a7d1bae9d79d4e672383c48f2e07116f4061 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:13 -0400
Subject: [ethernet] enic: remove unnecessary pci_set_drvdata()

Message-id: <1409668716-4814-2-git-send-email-sassmann@redhat.com>
Patchwork-id: 90628
O-Subject: [RHEL7.1 PATCH 01/24] net: enic: remove unnecessary pci_set_drvdata()
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Jingoo Han <jg1.han@samsung.com>

The driver core clears the driver data to NULL after device_release
or on probe failure. Thus, it is not needed to manually clear the
device driver data to NULL.

Signed-off-by: Jingoo Han <jg1.han@samsung.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 6d2d83e6316611ca8aa6daa9919c41398ba2725d)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/cisco/enic/enic_main.c b/drivers/net/ethernet/cisco/enic/enic_main.c
index 7b756cf..ff78dfa 100644
--- a/drivers/net/ethernet/cisco/enic/enic_main.c
+++ b/drivers/net/ethernet/cisco/enic/enic_main.c
@@ -2309,7 +2309,6 @@ err_out_release_regions:
 err_out_disable_device:
  pci_disable_device(pdev);
 err_out_free_netdev:
- pci_set_drvdata(pdev, NULL);
  free_netdev(netdev);
 
  return err;
@@ -2338,7 +2337,6 @@ static void enic_remove(struct pci_dev *pdev)
   enic_iounmap(enic);
   pci_release_regions(pdev);
   pci_disable_device(pdev);
-  pci_set_drvdata(pdev, NULL);
   free_netdev(netdev);
  }
 }
-- 
1.7.1